### PR TITLE
Rl save policies debugging

### PR DESCRIPTION
**Pull request by kim-mskw**
so the old save policies function did not really use max_eval. It was always set back to -1000 when the learning role was wirtten new. I changed that now and added an early stopping criterium. 
Something works better but something weird as well, because the max avg metrik is now always 19.28 

This is the basis for the discussion with @nick-harder ick- tomorrow

### CODE REVIEW

**Comment by kim-mskw** [comment]
okay I know why it is always the same, because it somehow now always loads old policies in a dead folder. I am fixing that. This is terribly confusing.

**Comment by kim-mskw** [comment]
so there was a function in the learning_strategies that loaded policies and one in the learning_algo, leading to very weird behaviour after the load_scenario function. Basically in the eval run always an old policy from a folder was read. 

Technically we need the actor in the bidding strategy, because otherwise it is not initialized if we do not use learning, so learning mode off and hence no algo as well. This double initializing is weird though.

File: assume/reinforcement_learning/algorithms/matd3.py [hunk 1]
```diff
@@ -328,7 +328,7 @@ def update_policy(self):
             This function implements the TD3 algorithm's key step for policy improvement and exploration.
         """
 
-        logger.info(f"Updating Policy")
+        # logger.info(f"Updating Policy")
```
Comment (nick-harder) [inline assume/reinforcement_learning/algorithms/matd3.py:331]: please remove this change, I have changed it to a debug message in my commit

File: assume/scenario/loader_csv.py [hunk 2]
```diff
@@ -367,7 +368,7 @@ async def load_scenario_folder_async(
     if not learning_config.get("trained_policies_save_path"):
         learning_config[
             "trained_policies_save_path"
-        ] = f"./learned_strategies/{study_case}"
+        ] = f"./learned_strategies/{study_case}/last_policies"
```
Comment (nick-harder) [inline assume/scenario/loader_csv.py:371]: last_policies are set twice now, here and also in load functions of the learning strategies.

File: assume/reinforcement_learning/learning_role.py [hunk 3]
```diff
@@ -45,6 +45,8 @@ def __init__(
         self.buffer: ReplayBuffer = None
         self.obs_dim = learning_config["observation_dimension"]
         self.act_dim = learning_config["action_dimension"]
+        self.early_stopping_steps = learning_config["early_stopping_steps"]
```
Comment (nick-harder) [inline assume/reinforcement_learning/learning_role.py:48]: these should have some default value set by us to make things a bit easier for the user

File: compose.yml [hunk 4]
```diff
@@ -17,16 +17,16 @@ services:
       - TS_TUNE_MAX_CONNS=500
     volumes:
       # needed for normal image
-      #- ./assume-db:/var/lib/postgresql/data
+      - ./assume-db:/var/lib/postgresql/data
```
Comment (nick-harder) [inline compose.yml:20]: why was the docker file changed?

File: examples/inputs/example_02a/config.yaml [hunk 5]
```diff
@@ -49,7 +49,7 @@ tiny:
 
 base:
   start_date: 2019-03-01 00:00
-  end_date: 2019-04-01 00:00
+  end_date: 2019-03-02 00:00
```
Comment (nick-harder) [inline examples/inputs/example_02a/config.yaml:52]: this was changed probably only for testing or?

File: compose.yml (hunk 4 above)
Comment (kim-mskw) [reply compose.yml:20]: Oh sorry, for my laptop the other image never works, I commited it accidently

**Pull request merged**